### PR TITLE
fix(tsm1): fix data race when accessing tombstone stats

### LINKED ISSUE #20622: Data race between FileStore and Tombstoner in tsm1
Exposed by CI [here](https://app.circleci.com/pipelines/github/influxdata/influxdb/18480/workflows/60ec696e-5e42-47f7-94d5-8a33b76cde93/jobs/171753). The race was:
```
WARNING: DATA RACE
Write at 0x00c00010ef60 by goroutine 151:
  github.com/influxdata/influxdb/v2/tsdb/engine/tsm1.(*Tombstoner).TombstoneFiles()
      /home/circleci/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/tombstone.go:216 +0x430
  github.com/influxdata/influxdb/v2/tsdb/engine/tsm1.(*TSMReader).TombstoneFiles()
      /home/circleci/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/reader.go:547 +0x6b
  github.com/influxdata/influxdb/v2/tsdb/engine/tsm1.(*FileStore).CreateSnapshot()
      /home/circleci/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/file_store.go:1087 +0x722
  github.com/influxdata/influxdb/v2/tsdb/engine/tsm1.(*Engine).CreateSnapshot()
      /home/circleci/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:1883 +0x1ce
  github.com/influxdata/influxdb/v2/tsdb.(*Shard).CreateSnapshot()
      /home/circleci/go/src/github.com/influxdata/influxdb/tsdb/shard.go:1141 +0xa6
  github.com/influxdata/influxdb/v2/tsdb_test.TestShard_WritePoints_FieldConflictConcurrent.func2()
      /home/circleci/go/src/github.com/influxdata/influxdb/tsdb/shard_test.go:434 +0x1ca

Previous read at 0x00c00010ef60 by goroutine 404:
  github.com/influxdata/influxdb/v2/tsdb/engine/tsm1.(*FileStore).CreateSnapshot()
      /home/circleci/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/file_store.go:1087 +0x7ca
  github.com/influxdata/influxdb/v2/tsdb/engine/tsm1.(*Engine).CreateSnapshot()
      /home/circleci/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:1883 +0x1ce
  github.com/influxdata/influxdb/v2/tsdb.(*Shard).CreateSnapshot()
      /home/circleci/go/src/github.com/influxdata/influxdb/tsdb/shard.go:1141 +0xa6
  github.com/influxdata/influxdb/v2/tsdb_test.TestShard_WritePoints_FieldConflictConcurrent.func1()
      /home/circleci/go/src/github.com/influxdata/influxdb/tsdb/shard_test.go:418 +0x1a4

Goroutine 151 (running) created at:
  github.com/influxdata/influxdb/v2/tsdb_test.TestShard_WritePoints_FieldConflictConcurrent()
      /home/circleci/go/src/github.com/influxdata/influxdb/tsdb/shard_test.go:425 +0xe97
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1123 +0x202

Goroutine 404 (running) created at:
  github.com/influxdata/influxdb/v2/tsdb_test.TestShard_WritePoints_FieldConflictConcurrent()
      /home/circleci/go/src/github.com/influxdata/influxdb/tsdb/shard_test.go:409 +0xe31
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1123 +0x202
```

### PR DESCRIPTION
**Pull request by danxmoran**
Closes #20622

### CODE REVIEW

File: tsdb/engine/tsm1/tombstone.go [hunk 1]
```diff
@@ -183,43 +182,51 @@ func (t *Tombstoner) Delete() error {
...
 // HasTombstones return true if there are any tombstone entries recorded.
 func (t *Tombstoner) HasTombstones() bool {
-	files := t.TombstoneFiles()
+	stats := t.TombstoneStats()
+	if !stats.TombstoneExists {
+		return false
+	}
+	if stats.Size > 0 {
+		return true
+	}
```
Comment (danxmoran) [inline tsdb/engine/tsm1/tombstone.go:191]: Not sure if it's worth the verbosity to avoid the extra lock below

Comment (benbjohnson) [reply tsdb/engine/tsm1/tombstone.go:191]: I think it's good as long as a non-zero size always means there are tombstones.

**Review by benbjohnson — APPROVED** [review]

File: tsdb/engine/tsm1/tombstone.go (hunk 1 above)
Comment (danxmoran) [reply tsdb/engine/tsm1/tombstone.go:191]: It's the same end behavior as before, at least

**Review by dgnorton — APPROVED** [review]

**Review by docmerlin — COMMENTED** [review]
LGTM

**Pull request merged**